perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourcesManagerImpl.java

```diff
@@ -514,4 +504,13 @@ public boolean extSourceExists(PerunSession perunSession, ExtSource extSource) t
 	public void checkExtSourceExists(PerunSession perunSession, ExtSource es) throws InternalErrorException, ExtSourceNotExistsException {
 		if(!extSourceExists(perunSession, es)) throw new ExtSourceNotExistsException("ExtSource: " + es);
 	}
+
+	public Map<String,String> getAttributes(ExtSource extSource) {
+		Map<String, String> attributes = new HashMap<>();
+		SqlRowSet srs = jdbc.queryForRowSet("select attr_name, attr_value from ext_sources_attributes where ext_sources_id = " + extSource.getId());
```

Comment:
Using ``queryForRowSet()`` is unsafe for HSQL DB (used in tests), since it by default doesn't publish renamed column names. I know, columns are not renamed in this sql query, but we should use standard ``jdbc.query()``. You can select multiple-lines as one object using own ``ResultSetExtractor`` and implementing its ``extractData()`` method. See e.g. ``AttributeRightsExtractor`` defined inside ``AttributesManagerImpl``. Thanks.